peps/pep-9999.rst

```diff
@@ -105,27 +115,83 @@ interval math or numerical integration.
+
+This PEP proposes a new module with only the six existing functions listed
+above.  However, in order to support the module's possible future utility, and
+to illustrate its scope, we provide an exploratory list of functions that could
+one day find a home in the ``imath`` library:
+
+* Number theory
+
+  * Divisibility & Modular Arithmetic (see also [5]_)
+
+    * ``gcd(a, b)``: Part of proposal.
+    * ``lcm(a, b)``: Part of proposal.
+    * ``powmod(a, b, m)``: Modular exponentiation (already in `pow(a, b, m)`,
+      but renamed for clarity).
+    * ``is_coprime(a, b)``
+    * ``modinv(a, m)``: Modular inverse.
+    * ``gcdext(a, b)``: Solve linear Diophantine equations in two variables (the
+      :external+py3.14:class:`int` implementation actually includes the extended
+      Euclidean algorithm).
+
+  * Integer Arithmetic and Sequences
+
+    * ``isqrt(n)``: Part of proposal.
+    * ``c_div(a, b)``: Ceiling divide, see [2]_, [3]_.
+    * ``isqrt_rem(n)``: The integer square root and its remainder.
+    * ``sqrtmod(n, modulo)``: Square root within a modular group.
+    * ``ilog(n, base)``: Integer logarihm; currently
+      :external+py3.14:func:`math.log` has a special handling for integer
+      arguments.  It's unique (wrt other module functions) and undocumented.
+      See [4]_.
+    * ``fibonacci(n)``
+
+  * Primality & Factorization
+
+    * ``is_prime(n)``
+    * ``is_power_of_two(x)``
+    * ``prime_factors(n)``: Trial division or Pollard’s rho for small inputs.
+    * ``totient(n)``: Euler’s totient function.
+
+* Combinatorics
+
+  * ``factorial(n)``: Part of proposal.
+  * ``comb(n, k)``: Part of proposal.
+  * ``perm(n, k)``: Part of proposal.
+
+
+Rejected ideas
+==============
+
+There was a brief discussion about exposing :external+py3.14:func:`math.isqrt`
+as ``imath.sqrt`` in the same way that :external+py3.14:func:`cmath.sqrt` is the
+complex version of :external+py3.14:func:`math.sqrt`. However, ``isqrt`` is
+ultimately a different function: it is the floor of the square root.  It would
+be confusing to give it the same name (under a different module).
+
+
+Arguments against
+=================
+
+Discoverability
+---------------
 
-Module scope and possible extensions
-------------------------------------
+Beginners looking for a math function may instinctively look for it in the math
+module.  Rather than creating a new module, change the documentation for the
+math module to read not mention the C standard or floating-point numbers.
 
-Unless we can just provide bindings to some well supported mathematical library
-like the GMP, the module scope should be limited.  For example, no primality
-testing and factorization.
+Countering this, complex math is still math, but we have a separate module for
+that.  Just as with complex numbers, integers should have their own module too.
 
-There are possible additions, among proposed in the initial discussion thread
-(see also [5]_):
+Limited benefit
+---------------
```

Comment:
I don't see counter arguments here.

Reply:
Should we add one?  I thought that our counterargument is that it is useful enough for all the reasons we've already given.  Or should we remove this section?  Happy with whatever.